eventsourcing/src/main/java/org/axonframework/eventsourcing/eventstore/ConsistencyMarker.java

```diff
@@ -74,4 +74,15 @@ public interface ConsistencyMarker {
      * @return a ConsistencyMarker that represents the upper bound of two other markers
      */
     ConsistencyMarker upperBound(ConsistencyMarker other);
+
+    /**
+     * Adds the given {@code consistencyMarker} to the given {@code context} using the {@link #RESOURCE_KEY}.
+     *
+     * @param context           The {@link Context} to add the given {@code token} to.
+     * @param consistencyMarker The {@code consistencyMarker} to add to the given {@code context} using the
+     *                          {@link #RESOURCE_KEY}.
+     */
+    static Context addToContext(Context context, ConsistencyMarker consistencyMarker) {
```

Comment:
I'm curious what the consideration is to make this a static method with an instance as parameter, instead of an instance method.

Reply:
I anticipated the use of a single resource key for this purpose among any implementation of (when I added it) the `TrackingToken`. Figured it would be easier to make a single `static` method on the `TrackingToken` instead of spreading out the same logic around to all `TrackingToken` instances.

Whether this style is applicable for **all** resources we envision is an entirely different point. See the `EventStoreTransaction#appendPosition` method for example. Nothing states it's a resource on the context, but for now it always is.

Anyhow, fully up to debate to change this around if you want. At least at the time, this felt like the least redundant solution I could conjure up.